pkg/discovery/kv/kv.go

```diff
@@ -63,9 +63,23 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 	s.ttl = ttl
 	s.path = path.Join(s.prefix, discoveryPath)
 
+	var config *store.Config
```

Comment:
Please see if you can use the TLSConfig package here: https://github.com/docker/docker/blob/master/pkg/tlsconfig/config.go

It does a good job of setting defaults (ciphersuite options, etc)